IdempotencyTests is flaky on Windows

Starting ~~with the merge of https://github.com/lampepfl/dotty/pull/11210~~ (EDIT: actually long before that, see comments below), it looks like test_windows_full has been failing IdempotencyTests from time to time, from https://github.com/lampepfl/dotty/runs/2178741539:
```scala
Test 'CheckPosIdempotency from idempotency/check' failed with output:           
Idempotency test failed between out\idempotency\posIdempotency1\pos\i6507b\Test.tasty and out\idempotency\posIdempotency2\pos\i6507b\Test.tasty
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at dotty.tools.vulpix.ChildJVMMain.runMain(ChildJVMMain.java:40)
	at dotty.tools.vulpix.ChildJVMMain.main(ChildJVMMain.java:47)
Caused by: java.lang.AssertionError: assertion failed: Failed 1 idempotency checks (out of 5116)
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
	at IdempotencyCheck$.checkIdempotency(IdempotencyCheck.scala:70)
	at Test$.main(CheckPosIdempotency.scala:4)
	at Test.main(CheckPosIdempotency.scala)
	... 6 more
```
@bishabosha could you investigate this (I've forwarded you the access information for the Windows machine if you want to investigate there)? /cc @liufengyun

--- a/tests/idempotency/IdempotencyCheck.scala
+++ b/tests/idempotency/IdempotencyCheck.scala
@@ -6,13 +6,12 @@ import java.util.stream.Stream as JStream
 import scala.collection.JavaConverters.*
 
 object IdempotencyCheck {
-  val blacklisted = Set(
-    // No fix needed. Bridges on collections in different order. Second one in scala2 order.
-    s"pos{JFile.separator}Map{JFile.separator}scala{JFile.separator}collection{JFile.separator}immutable/Map",
-    s"pos{JFile.separator}Map{JFile.separator}scala{JFile.separator}collection{JFile.separator}immutable{JFile.separator}AbstractMap",
-    s"pos{JFile.separator}t1203a/NodeSeq",
-    s"pos{JFile.separator}i2345{JFile.separator}Whatever"
-  )
+  val flakyTestsOnWindows =
+    if scala.util.Properties.isWin
+    then Set(s"pos${JFile.separator}i6507b")
+    else Set.empty
+
+  val blacklisted = flakyTestsOnWindows
 
   def checkIdempotency(dir1: String, dir2: String): Unit = {
     var failed = 0

